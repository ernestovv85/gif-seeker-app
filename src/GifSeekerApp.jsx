import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
const GifSeekerApp = () => {
  const [categories, setCategories] = useState([])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }
  return (
    <>
      <h1>Gif Seeker App</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  )
}

export default GifSeekerApp
