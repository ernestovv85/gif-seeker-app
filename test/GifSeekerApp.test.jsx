/* eslint-disable no-undef */

import { render, screen } from "@testing-library/react"
import GifSeekerApp from "../src/GifSeekerApp"

describe('Pruebas en <GifSeekerApp />', () => {
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifSeekerApp />)
    expect(container).toMatchSnapshot()
  })

  test('Debe mostrar el tÃ­tulo en el componente', () => {
    render(<GifSeekerApp />)
    expect(screen.getByText('Gif Seeker App')).toBeTruthy()
  })
})
