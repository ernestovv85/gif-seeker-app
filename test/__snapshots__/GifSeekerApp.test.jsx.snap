// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifSeekerApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Gif Seeker App
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gif"
      type="text"
      value=""
    />
  </form>
</div>
`;
